### DISCUSSION TITLE
[WIP] Privacy and Analytics

### DISCUSSION THREAD

**Discussion by vHanda** [GitJournal/GitJournal#824] [Feature Requests]
Currently in GitJournal, analytics is enabled by default, and can be disabled. I've gotten emails from a few users who have disabled analytics - as they weren't sure what was being collected, and when it was being sent (battery considerations).

As a developer of GitJournal, I - of course - want as much analytics as possible, it's a great positive feedback loop and helps guide development of features. However, as a User of an application, I always want the option to choose.

Currently, it's a binary option - on or off.

My idea is to turn it into a tiered system - where the user can choose between different levels of analytics. The lowest could just be if they used the app one day. Additionally, to provide a screen which shows all the analytics data which has been sent. I'm imagining a list of events. This way, the user is acutely aware of what is being tracked, instead of having to go to the privacy policy page, which is far more vague.

For very privacy oriented users, there could be an option to audit the events before sending them, this way - once a week, they could see all the analytics events GitJournal wants to send and accordingly choose.

This issue is to gather feedback about this idea, and for others to propose other ideas on being more transparent with analytics.

---
> Please vote on this issue with a 👍reaction. It's one of the primary mechanisms used to prioritize issues.

**Comment by aepound** [comment]
Very cool idea. Though it seems like it might turn into a very advanced type of configuration.
Would it be prudent to have a "simple" interface and an "advanced" one: the simple is on/off, and the advanced allows for the type of control that is described above?

**Comment by vHanda** [comment]
> Would it be prudent to have a "simple" interface and an "advanced" one: the simple is on/off, and the advanced allows for the type of control that is described above?

Definitely. The simple one shouldn't disappear.

**Comment by shreyasminocha** [comment]
I love the way KDE Plasma approaches this.

<ol>
	<li><details><summary>"Disabled" (default)</summary><img alt="Disabled" src="https://user-images.githubusercontent.com/11537232/107863467-8998e280-6e4c-11eb-89ff-e852646f8487.png" /></details></li>
	<li><details><summary>"Basic system information"</summary><img alt="Basic system information — application version, compiler info, platform info, qpa info, qt version info" src="https://user-images.githubusercontent.com/11537232/107863471-8b62a600-6e4c-11eb-8ce4-ed7b9c8c3c17.png" /></details></li>
	<li><details><summary>"Basic system information and usage statistics"</summary><img alt="Basic system information and usage statistics — all of the above plus launches count and usage time" src="https://user-images.githubusercontent.com/11537232/107863474-8d2c6980-6e4c-11eb-82b7-9cd411361bae.png" /></details></li>
	<li><details><summary>"Detailed system information and basic usage statistics"</summary><img alt="Detailed system information and basic usage statistics — all of the above plus locale info, OpenGL info, panel count, screen params" src="https://user-images.githubusercontent.com/11537232/107863475-8e5d9680-6e4c-11eb-817b-1d45acdb7e12.png" /></details></li>
	<li><details><summary>"Detailed system information and usage statistics"</summary><img alt="Detailed system information and usage statistics — all of the above plus application source name" src="https://user-images.githubusercontent.com/11537232/107863476-8f8ec380-6e4c-11eb-9c64-2ce88bb7f243.png" /></details></li>
</ol>
